grammar.y

```diff
@@ -393,35 +383,86 @@ table_alias:
 ;
 
 join_clause:
-  table_expr JOIN table_expr join_constraint
+  table_expr join_op table_expr join_constraint
   {
     if $4 == nil {
-      $$ = &JoinTableExpr{LeftExpr: $1, JoinOperator: JoinStr, RightExpr: $3}
+      $$ = &JoinTableExpr{LeftExpr: $1, JoinOperator: $2, RightExpr: $3}
     } else {
       $4.LeftExpr = $1
-      $4.JoinOperator = JoinStr
+      $4.JoinOperator = $2
       $4.RightExpr = $3
```

Comment:
OK, had to look that `join_constraint` already returns a pre-baked `JoinTableExpr`. :+1: